### PR TITLE
BUGFIX: Adjust code leading to deprecation warnings with PHP 8.1

### LINKED ISSUE #2797: Deprecation warnings with PHP 8.1
### Description

When using PHP 8.1 the tests emit a lot of deprecation warnings.

#### Expected behavior

No warnings.

#### Actual behavior

Tons of warnings.

### Affected Versions

All current versions.

### PR DESCRIPTION
**Pull request by kdambekalns**
Fixes #2797

### CODE REVIEW

**Review by mficzel — COMMENTED** [review]
Can confirm that this works locally and removes the deprecation warnings in php 8.1

However this causes an error in the Neos Unit tests that should be addressed. Also the neos dev-dist has  39 other deprecation warnings (when running the unit tests in php 8.1)
```
1) Neos\Neos\Tests\Unit\Routing\BackendModuleRoutePartHandlerTest::matchFindsCorrectValues with data set "empty" ('', -1, null)
Failed asserting that false is identical to -1.

/var/www/html/Packages/Neos/Neos.Neos/Tests/Unit/Routing/BackendModuleRoutePartHandlerTest.php:74
```

I read the code and it seems to be all fine, it is a lot to chew. I would give :thumbsup: once the Neos Side is addressed.

File: Neos.Flow/Classes/Validation/Validator/TextValidator.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ class TextValidator extends AbstractValidator
      */
     protected function isValid($value)
     {
-        if ($value !== filter_var($value, FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES)) {
+        if ($value !== strip_tags((string)$value)) {
```
Comment (albe) [inline Neos.Flow/Classes/Validation/Validator/TextValidator.php:39]: FILTER_FLAG_NO_ENCODE_QUOTES should be replaced with https://www.php.net/manual/de/function.htmlspecialchars.php - so should we do that as well here?

Comment (kdambekalns) [reply Neos.Flow/Classes/Validation/Validator/TextValidator.php:39]: Tried that first, but for some reason a test failed then. As per the method description "text without any XML tags" is to be considered valid, so I figured `strip_tags()` should od the job – and the tests pass.

**Comment by kdambekalns** [comment]
> However this causes an error in the Neos Unit tests that should be addressed. Also the neos dev-dist has 39 other deprecation warnings (when running the unit tests in php 8.1)

Yes, this PR focuses solely on Flow, Neos follows…

**Review by mficzel — APPROVED** [review]
Can confirm that this works also locally on PHP 8.0 and 8.1.

Changes are looking good but i did not really reflect each line deeply.

**Comment by jonnitto** [comment]
Question @kdambekalns: There are quite a few functions that do not have a return value declaration. Are these intentionally omitted?

**Comment by kdambekalns** [comment]
> There are quite a few functions that do not have a return value declaration. Are these intentionally omitted?

Main intention was to fix deprecation wanrings, I did refrain from changing/adding type declarations as much as I could…

So: Yes.

**Pull request merged**